external-videos.php

```diff
@@ -259,26 +255,25 @@ function admin_scripts( $hook ) {
   static function get_options(){
 
     // Load existing plugin options
-    $raw_options = get_option( 'sp_external_videos_options' );
+    $options = get_option( 'sp_external_videos_options' );
     // echo '<pre>$raw_options: '; print_r($raw_options); echo '</pre>';
 
+    if( !$options ) {
+      $options = array();
+    }
+    // Upgrade options to v1.0: move author options to hosts
+    $options = SP_External_Videos::convert_author_options( $options );
+
     // Set defaults for the basic options
-    if( !$raw_options ) {
-      $options = array( 'version' => 1, 'rss' => false, 'delete' => true, 'hosts' => array(), 'attrib' => false, 'loop' => false, 'slug' => 'external-videos');
```

Comment:
This I tested - it is OK to remove actually. 

At this point, for first time initialization, $options is set a blank array (line 262), and the isset() statements below set our defaults one key at a time. The reason one-at-a-time is preferable is for upgrades - people with earlier versions will have some of the keys set, but not the new ones. So upgraders don't get the defaults unless it checks for them one by one.

But definitely, first time users get all defaults set by the isset() statements. I just tested on a fresh install.